examples/raja-teams-tile.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Define thread policies.
+ * Up to 3 dimension are supported: x,y,z
+ */
+using threads_x = RAJA::expt::LoopPolicy<RAJA::loop_exec
+#if defined(RAJA_ENABLE_CUDA)
+                                         ,
+                                         RAJA::cuda_thread_x_loop
+#endif
+#if defined(RAJA_ENABLE_HIP)
+                                         ,
+                                         RAJA::hip_thread_x_loop
+#endif
+                                         >;
+
+
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  // Resource object for host
+  camp::resources::Host host_res;
+
+  // Resource objects for CUDA or HIP
+#if defined(RAJA_ENABLE_CUDA)
+  camp::resources::Cuda device_res;
+#endif
+
+#if defined(RAJA_ENABLE_HIP)
+  camp::resources::Hip device_res;
+#endif
+
+  std::cout << "\n Running RAJA-Teams examples...\n";
+  int num_of_backends = 1;
+#if defined(RAJA_DEVICE_ACTIVE)
+  num_of_backends++;
+#endif
+
+  // RAJA teams may switch between host and device policies at run time.
+  // The loop below will execute through the available backends.
+
+  for (int exec_place = 0; exec_place < num_of_backends; ++exec_place) {
+
+    RAJA::expt::ExecPlace select_cpu_or_gpu = (RAJA::expt::ExecPlace)exec_place;
+
+    // auto select_cpu_or_gpu = RAJA::HOST;
+    // auto select_cpu_or_gpu = RAJA::DEVICE;
+
+    // Allocate memory for either host or device
+    int N_tri = 10;
+
+    int* Ddat = nullptr;
+    if (select_cpu_or_gpu == RAJA::expt::HOST) {
+      Ddat = host_res.allocate<int>(N_tri * N_tri);
+    }
+
+#if defined(RAJA_DEVICE_ACTIVE)
+    if (select_cpu_or_gpu == RAJA::expt::DEVICE) {
+      Ddat = device_res.allocate<int>(N_tri * N_tri);
+    }
+#endif
+
+    /*
+     * RAJA::expt::launch just starts a "kernel" and doesn't provide any looping.
+     *
+     * The first argument determines which policy should be executed,
+     *
+     * The second argument is the number of teams+threads needed for each of the
+     * policies.
+     *
+     * Third argument is the lambda.
+     *
+     * The lambda takes a "resource" object, which has the teams+threads
+     * and is used to perform thread synchronizations within a team.
+     */
+
+    if (select_cpu_or_gpu == RAJA::expt::HOST){
+      std::cout << "\n Running upper triangular pattern example on the host...\n";
+    }else {
+      std::cout << "\n Running upper triangular pattern example on the device...\n";
+    }
+
+
+    RAJA::View<int, RAJA::Layout<2>> D(Ddat, N_tri, N_tri);
+
+    RAJA::expt::launch<launch_policy>(select_cpu_or_gpu,
+       RAJA::expt::Resources(RAJA::expt::Teams(N_tri), RAJA::expt::Threads(N_tri)),
+       [=] RAJA_HOST_DEVICE(RAJA::expt::LaunchContext ctx) {
+
+        const int TILE_SZ = 4;
+        RAJA::expt::tile_idx<teams_x>(ctx, TILE_SZ, RAJA::RangeSegment(0, N_tri), [&](RAJA::RangeSegment const &r_tile, int tile_id) {
+
+             RAJA::expt::loop_idx<loop_t>(ctx, r_tile, [&](int global_id, int loc_id) {
```

Comment:
@artv3 do we want to call this loop_icount to be consistent with forall_icount?

Reply:
Good suggestion 👍

Reply:
Yes, that is a good idea.